File: Task_Scheduler.py
class Solution:
    def leastInterval(self, N, K, tasks):
        # Code here
        c={}
        for x in tasks :
            c[x] = c.get(x,0) + 1
        m=0
        for x in c :
            m=max(m,c[x])
        cnt=0
        for x in c :
            if c[x]==m :
                cnt +=1
        ans = max(N, ((m-1)*(1+K) + cnt))
        return ans

File: Task_Scheduler.cpp
int leastInterval(int N, int K, vector<char> &tasks) {
        
        map<char, int> m;
        for(int i=0;i<N; i++){
            m[tasks[i]]++;
        }
        int mx=INT_MIN;
        for(auto i: m){
            if(i.second > mx){
                mx = i.second;
            }
        }
        int ideal = (mx-1)*K;
        ideal+=(mx-1);
        for(auto x : m){
            ideal-=min(mx-1, x.second);
        }
        return tasks.size() + max(0,ideal);
    }
